### PR TITLE
[Tests] Update test-gpio.js for bug fix

### PR DESCRIPTION
**Pull request by cuiyanx**
Signed-off-by: Cui, Yan yanx.cui@intel.com

### CODE REVIEW

**Comment by haoxli** [comment]
@cuiyanx  I run the tests but get 21 tests result, I remember there should be 15 tests totally, please double check the tests number.

File: tests/test-gpio.js [hunk 1]
```diff
@@ -73,51 +73,43 @@ bValue = pinB.read();
...
+            if(key == "any") {
+                // test GPIOPin close
+                pinA.write(!inital);
+                assert(count == changes[key], "gpiopin: close onchange");
+
+                testChangeWithEdge("rising", false);
+            } else if(key == "rising") {
+                testChangeWithEdge("falling", true);
+            } else if(key == "falling") {
+                console.log("TOTAL: " + passed + " of " + total + " passed");
```
Comment (grgustaf) [inline tests/test-gpio.js]: TOTAL gets printed multiple times

**Comment by cuiyanx** [comment]
I test this case and have 15/15 pass, please use "wire IO2 to IO4" and "make JS=tests/test-gpio.js MALLOC=heap".

**Comment by grgustaf** [comment]
This isn't a big deal here, but @cuiyanx and @haoxli we prefer everyone to submit patch review changes as small additional commits now, instead of force pushing. That way we can look at the commits showing what changed since the last time, instead of reviewing the whole thing again.

**Comment by grgustaf** [comment]
@cuiyanx I run the same make command you gave and the tests pass until here:

```
PASS - gpiopin: close onchange 
FAIL - gpiopin: onchange with edge 'rising' 
FAIL - gpiopin: onchange with edge 'rising' 
FAIL - gpiopin: onchange with edge 'rising' 
FAIL - gpiopin: onchange with edge 'rising' 
FAIL - gpiopin: onchange with edge 'falling' 
TOTAL: 13 of 18 passed 
FAIL - gpiopin: onchange with edge 'falling' 
TOTAL: 13 of 19 passed 
FAIL - gpiopin: onchange with edge 'falling' 
TOTAL: 13 of 20 passed 
PASS - gpiopin: onchange with edge 'falling' 
TOTAL: 14 of 21 passed 
```

It doesn't seem to make a difference whether I use MALLOC=heap or not.

**Comment by haoxli** [comment]
Got it. We will submit additional commits for changes.

**Comment by grgustaf** [comment]
Interesting, it does pass 15/15 for me now.

But yesterday as I was looking at it, I noticed that the gpio pins were getting reopened without being closed. When I started adding close() calls, it was hanging inside openAsync() and other weird behavior, which seemed interesting. It may have been revealing some real bugs.

**Comment by grgustaf** [comment]
Anyway, that might be worth looking at more, but I'll merge this now.

+1

**Pull request merged**